pkg/emailer/service_test.go

```diff
@@ -66,3 +67,49 @@ func Test_Email_job_push_error(t *testing.T) {
 
 	assert.EqualError(t, err, "some-error")
 }
+
+func TestSendCampaignEmail(t *testing.T) {
+	brokerMock := job.NewBrokerMock(t)
+
+	emailer := NewEmailerService(brokerMock)
+
+	inst := instance.Instance{}
+
+	t.Run("WithoutSubject", func(t *testing.T) {
+		err := emailer.SendCampaignEmail(&inst, &CampaignEmailCmd{})
+		assert.ErrorIs(t, err, ErrMissingSubject)
+	})
+
+	t.Run("WithoutParts", func(t *testing.T) {
+		err := emailer.SendCampaignEmail(&inst, &CampaignEmailCmd{
+			Subject: "Some subject",
+		})
+		assert.ErrorIs(t, err, ErrMissingContent)
+	})
+
+	t.Run("WithCompleteCmd", func(t *testing.T) {
+		brokerMock.On("PushJob", &inst, mock.MatchedBy(func(req *job.JobRequest) bool {
+			assert.Equal(t, "sendmail", req.WorkerType)
+			assert.JSONEq(t, `{
+    "mode": "campaign",
+	"subject": "Some subject",
```

Comment:
The indentation looks weird here

Reply:
I let the tools deal with this usually, even if the result is not to my taste.
But the tests pass with a more readable indentation so let's go with that :).

Reply:
If you are speaking of `gofmt`, it doesn't format what is inside a string